response_operations_ui/views/respondents.py

```diff
@@ -50,4 +50,91 @@ def respondent_details(respondent_id):
     ]
 
     respondent['status'] = respondent['status'].title()
-    return render_template('respondent.html', respondent=respondent, enrolments=enrolments, breadcrumbs=breadcrumbs)
+
+    info_message = None
```

Comment:
This whole block feels messy.  Have you looked into using Flash (http://flask.pocoo.org/docs/1.0/patterns/flashing/), it might be a cleaner way to do the same thing

Reply:
Also, as with the below comment, having lots of if statements that keep changing the message is asking for weird bugs to occur at some point.  There's probably a better way to do it (off the top of my head, using if/elif might be the simplest one, but there might be cleaner ways to do it)

Reply:
Thinking about this more, is probably the way to do this.  You can flash multiple times, then in the template you can loop and render every flashed message.  
This way, if there are multiple things to report on, you'll see them all as opposed to just the last one

Reply:
added now